### PR TITLE
You can corrupt generators by clicking on it

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Remove the corrupt generators ability, you know just have to click on it as a queen or shrike

## Why It's Good For The Game

Less button is good

## Changelog
:cl:
qol: You can corrupt a generator as queen or shrike by clicking on it. Corrupt generators ability is gone
/:cl:

### CODE REVIEW

File: code/modules/power/groundmap_geothermal.dm [hunk 1]
```diff
@@ -137,7 +137,16 @@
 
 /obj/machinery/power/geothermal/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
 	. = ..()
-	// Can't damage a broken generator
+	if(isxenoqueen(X)||isxenoshrike(X))
+		if(!is_corruptible)
```
Comment (TiviPlus) [inline code/modules/power/groundmap_geothermal.dm:141]: This should be a caste flag

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 2]
```diff
@@ -83,6 +83,9 @@
 	///whether or not a caste can hold eggs, and either 1 or 2 eggs at a time.
 	var/can_hold_eggs = CANNOT_HOLD_EGGS
 
+	///whether or not a caste can corrupt a generator
+	var/can_corrupt_gene = FALSE
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:87]: caste_flags ??

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 2 above)
Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:87]: aaah, yes, my bad

File: code/__DEFINES/mobs.dm [hunk 3]
```diff
@@ -477,6 +477,7 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
 #define CASTE_ACID_BLOOD (1<<13) //The acid blood effect which damages humans near xenos that take damage
 #define CASTE_CAN_HOLD_JELLY (1<<14)//whether we can hold fireproof jelly in our hands
 #define CASTE_IS_STRONG (1<<15)//can tear open acided walls without being big
+#define CASTE_CAN_CORRUPT_GENE (1<<16) //Can corrupt a generator
```
Comment (TiviPlus) [inline code/__DEFINES/mobs.dm:480]:
```suggestion
#define CASTE_CAN_CORRUPT_GENERATOR (1<<16) //Can corrupt a generator
```
the other one isnt that clear

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**